crates/solver/src/settlement_submission.rs

```diff
@@ -357,20 +384,26 @@ impl SolutionSubmitter {
             strategy_args.sub_tx_pool.clone(),
             self.web3.clone(),
             self.code_fetcher.as_ref(),
-        )?;
-        submitter
-            .submit(settlement, params)
-            .await
-            .map(|tx| SubmissionReceipt {
+        )
+        .map_err(|err| SubmissionError {
+            inner: SubmissionErrorClass::from(err),
+            strategy: strategy.label(),
+        })?;
+        match submitter.submit(settlement, params).await {
+            Ok(tx) => Ok(SubmissionReceipt {
                 tx,
                 strategy: strategy.label(),
-            })
+            }),
+            Err(inner) => Err(SubmissionError {
+                inner,
+                strategy: strategy.label(),
+            }),
+        }
     }
 }
 
-/// An error during settlement submission.
 #[derive(Debug)]
-pub enum SubmissionError {
+pub enum SubmissionErrorClass {
```

Comment:
Nit: Not sure renaming is super appropriate here.
1. This enum is the actual error and the new struct is an error + meta information so I would rather give that one a different name.
2. Way bigger diff